### PR TITLE
Make xz compression use all available cores

### PR DESCRIPTION
**Pull request by XenonPK**
Compression takes a while.  This should make it faster. (Tested on my local machine)

### CODE REVIEW

File: recompress [hunk 1]
```diff
@@ -78,7 +78,7 @@ for i in $FILES; do
     COMPRESS="bzip2 -c -"
     NEWFILE="${BASENAME#_service:}.bz2"
   elif [ "$MYCOMPRESSION" == "xz" ]; then
-    COMPRESS="xz -c -"
+    COMPRESS="xz --threads=$(nproc --ignore=1) -c -"
     NEWFILE="${BASENAME#_service:}.xz"
```
Comment (M0ses) [inline recompress]: Why not simply setting threads to 0? From `man xz`:

> Setting threads to a special value 0 makes xz use as many threads as there are CPU cores on the system.

Comment (XenonPK) [inline recompress]: Setting threads to 0 works (just tested it), and is simpler, it just did not cross my mind at the time. I'll change it.

**Comment by M0ses** [comment]
Hmm, 3 commits for 1 line changed - could you sqash them?

**Comment by marxin** [comment]
Yes, `-T0` seems to me like a logic option.

**Comment by marxin** [comment]
Note that the same option can be used for `zstd` compression.

**Comment by bmwiedemann** [comment]
I found yesterday that our b4 package build became unreproducible because xz compression is different if 1 thread is used. https://lists.archlinux.org/pipermail/arch-dev-public/2019-March/029520.html also noticed this.

And here is a reproducer:
`for n in 2 3 ; do echo | taskset $n xz --threads=0 -c - | md5sum ; done`

Could we use `--threads=$(n=$(nproc); [[ $n > 1 ]] || n=2; echo $n)`

**Comment by marxin** [comment]
It's a known limitation of `xz` compression and that's why we want to use `zstd` that doesn't suffer from the problem.

**Comment by adrianschroeter** [comment]
the need to avoid threads=1 is not a reason to create larger files IMHO...

**Pull request merged**